subscription.go

```diff
@@ -43,5 +44,8 @@ func (sub *Subscription) Cancel() {
 }
 
 func (sub *Subscription) close() {
-	close(sub.ch)
+	if !sub.closed {
+		close(sub.ch)
+		sub.closed = true
+	}
```

Comment:
this needs to be atomic or we can have two concurrent closes fail; can you make this an swap/compare with an atomic variable?

Reply:
Looked like close is never called concurrently; only called by the handler goroutine. ...but in case it is someday...